### PR TITLE
Expeditionary corps + Black Mesa + Trusted Player System

### PR DESCRIPTION
**Pull request by Gandalf2k15**
## Welcome to Expeditionary corps.

This expansion includes:
A new job - Expeditionary Trooper
The Expeditionary Trooper is expected to be at the forefront of exploration and discovery. They're given access to some advanced equipment and technology to help them along the way. You answer to the Captain, and the Captain only. You are NOT part of security and are forbidden from arresting people/acting like security. Due to the fact this role is quite abusable, it will be introduced as part of the trusted player system.
A bunch of new gateways - We've redone the old gateways and will be introducing new ones soon!

## Trusted Player System - TBA
Due to the fact we've had a few issues regarding players and allowing them the maximum freedom our server provides, we've decided to add a system for commending and rewarding our best players.

Currently we've renenabled the clown and restricted it under this system. If you are interested in joining the trusted player system, please see: TBA

# Black Mesa Content
We've added a new gateway map; black mesa. It's not easy, in fact, it's near impossible. 

A multitude of new BMS mobs have been added, along with a bit of BMS content.

A new BMS VOX system has been added too.

Credits to Crowbar Collective for any sound assets; specifically from the free Black Mesa Source mod.

Exp trooper clothing comes from Eris!

Bullsquid: https://github.com/HippieStation/HippieStation/pull/12825

## Changelog
:cl:
add: Expeditionary corps has been added!
add: The trusted player system is now in place, see: TBA
add: A new gateway map has been added, Black Mesa.
add: Lots of new Black Mesa content has been added.
add: A new VOX system has been added.
/:cl:

### CODE REVIEW

**Comment by KasparoVy** [comment]
Fucking awesome.

**Comment by FlamingLily** [comment]
YEAAAAAAAAAAH WOOOOOOO

**Comment by Arctaisia** [comment]
:eyes:

**Comment by Cheshify** [comment]
Ugh, this is a kick in the ass to get started on my gateway map. All in all, code looks fine to me? I'd still like it if we could use a seperate gunkit system for different Exp Corp loadouts, might add some depth. Want to add random cache spawns as well, but we'll see about that after they're implemented and looking good. All in all, very promising PR 👍

**Comment by FlamingLily** [comment]
Perhaps the blueshield should be added to the trusted players system? Because they too aren't sec and should not be behaving like sec ever.

**Comment by hugmeorelse** [comment]
> Perhaps the blueshield should be added to the trusted players system? Because they too aren't sec and should not be behaving like sec ever.

Great idea. 

Also, I want to add that Trusted Players isn't mean to be exclusionary. Rather, it's there as a reward for good, non-toxic players. We want as many players as possible in this category.

**Comment by thestubborn** [comment]
name a dumb clothing item* and i'll put it as a trusted player reward

*that has a sprite
i was thinking maybe a medal accessory

**Comment by FlamingLily** [comment]
yo what kinda alt titles you want on these bad boys?

Maybe like 
"Expeditionary Medic"
"Expeditionary Recon"
something like that?

Or do we wanna go full milsim, have ranks?
"Expeditionary Private"
"Expeditionary Corporal"

I dunno, spitballing

**Comment by KathrinBailey** [comment]
might be best to reword "trusted player" and all that jazz to whitelist?
Avoids drama since it sounds kinda favoritism-y.

**Comment by YakumoChen** [comment]
the corps armor values are sort of ridiculous, they're dangerously close to the armor values the _deathsquad_ gets. Hell, the armor has better laser protection than an actual Squad SWAT suit.

You should really lop like, 20 off those armor values, they seem overpowered and overly tanky in general.

**Comment by YakumoChen** [comment]
For reference, 60 Laser armor matches the Ablative vest, an armor _specifically_ designed exclusively to function against laser weaponry.

For further reference, these are the armor values for the cap's carapace, which is what should be the best station-side armor in the game.
	armor = list(MELEE = 50, BULLET = 40, LASER = 50, ENERGY = 50, BOMB = 25, BIO = 0, RAD = 0, FIRE = 100, ACID = 90, WOUND = 10)

**Comment by Gandalf2k15** [comment]
> For reference, 60 Laser armor matches the Ablative vest, an armor _specifically_ designed exclusively to function against laser weaponry.
> 
> For further reference, these are the armor values for the cap's carapace, which is what should be the best station-side armor in the game.
> armor = list(MELEE = 50, BULLET = 40, LASER = 50, ENERGY = 50, BOMB = 25, BIO = 0, RAD = 0, FIRE = 100, ACID = 90, WOUND = 10)

It's TBA, not really ready yet. We don't have any new gateways. This Is very much a WIP

**Comment by Gandalf2k15** [comment]
EXP corps are now considered science.

File: code/modules/modular_computers/file_system/programs/jobmanagement.dm [hunk 1]
```diff
@@ -24,10 +24,10 @@ GLOBAL_VAR_INIT(time_last_changed_position, 0)
...
 		"Research Director",
-		"Security Sergeant",
+		"Security Sergeant", //SKYRAT EDIT CHANGE START
 		"Security Medic",
 		"Clown",
-		"Blueshield",
+		"Blueshield", //SKYRAT EDIT END
 		"Chief Medical Officer",
 		"Quartermaster",
 		"Shuttle Pilot")
```
Comment (tf-4) [inline code/modules/modular_computers/file_system/programs/jobmanagement.dm:33]: Shuttle pilot should be in the skyrat edit (maybe QM? Not actually sure on that)

File: modular_skyrat/master_files/code/datums/id_trim/jobs.dm [hunk 2]
```diff
@@ -143,4 +154,3 @@
 				ACCESS_MAINT_TUNNELS, ACCESS_ENTER_GENPOP, ACCESS_LEAVE_GENPOP)
 
 	config_job = "brigoff"
-	template_access = list(ACCESS_CAPTAIN, ACCESS_HOS, ACCESS_CHANGE_IDS)
```
Comment (tf-4) [inline modular_skyrat/master_files/code/datums/id_trim/jobs.dm:146]: Why delete this?

File: modular_skyrat/modules/exp_corps/code/clothing.dm [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+	armor = list(MELEE = 60, BULLET = 60, LASER = 60, ENERGY = 60, BOMB = 60, BIO = 0, RAD = 20, FIRE = 80, ACID = 100, WOUND = 30)
+
+/obj/item/clothing/gloves/combat/expeditionary_corps
+	name = "expeditionary corps gloves"
+	icon_state = "exp_corps"
+	icon = 'modular_skyrat/modules/customization/icons/obj/clothing/gloves.dmi'
+	worn_icon = 'modular_skyrat/modules/customization/icons/mob/clothing/hands.dmi'
+
+/obj/item/storage/backpack/duffelbag/expeditionary_corps
+	name = "expeditionary corps bag" //SKYRAT EDIT, was "suspicious-looking duffel bag". It's just a black duffel.
```
Comment (tf-4) [inline modular_skyrat/modules/exp_corps/code/clothing.dm]:
```suggestion
	name = "expeditionary corps bag"
```

File: modular_skyrat/modules/exp_corps/code/clothing.dm [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+	inhand_icon_state = "sec_helm"
+	hardsuit_type = "hexp_corps"
+	armor = list(MELEE = 60, BULLET = 60, LASER = 60, ENERGY = 60, BOMB = 60, BIO = 0, RAD = 20, FIRE = 80, ACID = 100, WOUND = 30)
+	visor_flags_inv = HIDEMASK|HIDEEYES|HIDEFACE|HIDEFACIALHAIR|HIDESNOUT
+	visor_flags = STOPSPRESSUREDAMAGE
+	slowdown = 0.5
+
+/obj/item/clothing/head/helmet/expeditionary_corps
+	name = "expeditionary corps helmet"
+	desc = "A robust helmet worn by Expeditionary Corps troopers. Alt+click it to toggle the NV system."
```
Comment (tf-4) [inline modular_skyrat/modules/exp_corps/code/clothing.dm]: Tbh I'd like this on an action button, even if people dislike them. It's easier to access in combat

File: modular_skyrat/modules/exp_corps/code/clothing.dm [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+	if(nightvision)
+		var/obj/item/organ/eyes/my_eyes = current_user.getorgan(/obj/item/organ/eyes)
+		if(my_eyes)
+			my_eyes.lighting_alpha = initial(my_eyes.lighting_alpha)
+			my_eyes.see_in_dark = initial(my_eyes.see_in_dark)
+			my_eyes.flash_protect = initial(my_eyes.flash_protect)
+		current_user.remove_client_colour(/datum/client_colour/glass_colour/lightgreen)
+		current_user.update_sight()
+	current_user = null
+	return ..()
```
Comment (tf-4) [inline modular_skyrat/modules/exp_corps/code/clothing.dm]: Feels like unnecessary duplication

File: modular_skyrat/modules/exp_corps/code/expeditionary_trooper.dm [hunk 6]
```diff
@@ -0,0 +1,121 @@
...
+	paycheck_department = ACCOUNT_SCI
+
+	display_order = JOB_DISPLAY_ORDER_EXP_CORPS
+	bounty_types = CIV_JOB_SCI
+
+	family_heirlooms = list(/obj/item/binoculars)
+
+	trusted_only = TRUE
+
+/datum/job/expeditionary_trooper/after_spawn(mob/living/carbon/human/H, mob/M)
```
Comment (tf-4) [inline modular_skyrat/modules/exp_corps/code/expeditionary_trooper.dm:25]: https://github.com/tgstation/tgstation/pull/56872

File: modular_skyrat/modules/exp_corps/code/expeditionary_trooper.dm [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+
+/obj/item/storage/box/expeditionary_survival
+	name = "expedition survival pack"
+	desc = "A box filled with useful items for your expedition!"
+	icon_state = "survival_pack"
+	icon = 'modular_skyrat/modules/exp_corps/icons/survival_pack.dmi'
+	illustration = null
+
+/obj/item/storage/box/expeditionary_survival/PopulateContents()
+	new /obj/item/crowbar/red(src)
```
Comment (tf-4) [inline modular_skyrat/modules/exp_corps/code/expeditionary_trooper.dm]: This is redundant seeing as they get a full toolbelt

File: modular_skyrat/modules/exp_corps/code/expeditionary_trooper.dm [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+	new /obj/item/clothing/suit/space/hardsuit/expeditionary_corps(src)
+	new /obj/item/radio(src)
+	new /obj/item/melee/tomahawk(src)
+	new /obj/item/clothing/gloves/combat/expeditionary_corps(src)
+	new /obj/item/clothing/head/helmet/expeditionary_corps(src)
+	new /obj/item/clothing/suit/armor/vest/expeditionary_corps(src)
+	new /obj/item/storage/belt/military/expeditionary_corps(src)
+	new /obj/item/storage/backpack/duffelbag/expeditionary_corps(src)
+
+/obj/item/exp_corps_equip
```
Comment (tf-4) [inline modular_skyrat/modules/exp_corps/code/expeditionary_trooper.dm:99]: Why put everything in a beacon, and not just the hardsuit?

File: modular_skyrat/master_files/code/datums/id_trim/jobs.dm (hunk 2 above)
Comment (Gandalf2k15) [reply modular_skyrat/master_files/code/datums/id_trim/jobs.dm:146]: redundant

File: modular_skyrat/modules/exp_corps/code/expeditionary_trooper.dm (hunk 6 above)
Comment (Gandalf2k15) [reply modular_skyrat/modules/exp_corps/code/expeditionary_trooper.dm:25]: can't change how VSC sets it up, H/M is fine.

File: modular_skyrat/modules/exp_corps/code/clothing.dm (hunk 5 above)
Comment (Gandalf2k15) [reply modular_skyrat/modules/exp_corps/code/clothing.dm]: True

File: modular_skyrat/modules/exp_corps/code/expeditionary_trooper.dm (hunk 8 above)
Comment (Gandalf2k15) [reply modular_skyrat/modules/exp_corps/code/expeditionary_trooper.dm:99]: So people feel like they have to equip things.

**Comment by YakumoChen** [comment]
New maps are great and all but I cant help but feel you should sort out and merge the non-map content first, then PR the new maps

**Comment by YakumoChen** [comment]
Is using this much content from Half Life even legal? I know we have xenos and such in the code but these aren't even creatively renamed, it's literally just using Half Life trademarks.

**Comment by thestubborn** [comment]
> 
> 
> Is using this much content from Half Life even legal? I know we have xenos and such in the code but these aren't even creatively renamed, it's literally just using Half Life trademarks.

i mean neither are a lotta things

though I enjoy renaming/retitling stuff, I think blappeture mesa would be a fun rename, as it technically happened in a different universe

**Comment by YakumoChen** [comment]
> 
> 
> > Is using this much content from Half Life even legal? I know we have xenos and such in the code but these aren't even creatively renamed, it's literally just using Half Life trademarks.
> 
> i mean neither are a lotta things
> 
> though I enjoy renaming/retitling stuff, I think blappeture mesa would be a fun rename, as it technically happened in a different universe

care to name any? As I said, explicit trademark usage is a good way to invite a C&D, and pretty much every referenced thing I know about is either named away from the trademark or not a copyright issue.

**Comment by thestubborn** [comment]
> 
> 
> > > Is using this much content from Half Life even legal? I know we have xenos and such in the code but these aren't even creatively renamed, it's literally just using Half Life trademarks.
> > 
> > 
> > i mean neither are a lotta things
> > though I enjoy renaming/retitling stuff, I think blappeture mesa would be a fun rename, as it technically happened in a different universe
> 
> care to name any? As I said, explicit trademark usage is a good way to invite a C&D, and pretty much every referenced thing I know about is either named away from the trademark or not a copyright issue.

you said it yourself, xenomorphs, which ....are called aliens i stand corrected
we use some portal 2 voicelines for the SM blowing
the doug dimmadome suit that's dead ass named that, not c&d issue level but
ehh
just a few off the top of my head

**Comment by Gandalf2k15** [comment]
We should be okay as we're not using actual model assets. Black Mesa is not protected either. Some of these assets are from other servers too.

**Comment by NullDagaf** [comment]
> might be best to reword "trusted player" and all that jazz to whitelist?
> Avoids drama since it sounds kinda favoritism-y.

^
They're right, you know.

**Comment by Gandalf2k15** [comment]
Yeah these are free mod release assets. And are licensed under the Creativs Commons license. And we are not making any money off this, no profit.

And also; its from Black Mesa Source(free version)

**Comment by NullDagaf** [comment]
Can't find any mention of "Black Mesa" being copyrighted either.

**Comment by thestubborn** [comment]
black messa

**Comment by NullDagaf** [comment]
> Yeah these are free mod release assets. And are licensed under the Creativs Commons license. And we are not making any money off this, no profit.
> 
> And also; its from Black Mesa Soruce(free version)

SOURCE NOT SORUCE

**Comment by Gandalf2k15** [comment]
> > Yeah these are free mod release assets. And are licensed under the Creativs Commons license. And we are not making any money off this, no profit.
> > And also; its from Black Mesa Soruce(free version)
> 
> SOURCE NOT SORUCE

Phone moment

**Comment by NullDagaf** [comment]
> > > Yeah these are free mod release assets. And are licensed under the Creativs Commons license. And we are not making any money off this, no profit.
> > > And also; its from Black Mesa Soruce(free version)
> > 
> > 
> > SOURCE NOT SORUCE
> 
> Phone moment

okay better.

**Comment by NullDagaf** [comment]
>words first, reply later

github moment

**Comment by FlamingLily** [comment]
> 
> 
> We should be okay as we're not using actual model assets. Black Mesa is not protected either. Some of these assets are from other servers too.

We're also a non-profitable 18+ furry (E)RP server on a 2003 atmospherics simulator with less daily players than the original half life, not even counting spinoffs, mods, sequels, etc.

I think we'll be fine.

File: code/__DEFINES/jobs.dm [hunk 9]
```diff
@@ -14,44 +14,44 @@
...
+#define JOB_DISPLAY_ORDER_EXP_CORPS 32 //SKYRAT EDIT ADDITON
+#define JOB_DISPLAY_ORDER_HEAD_OF_SECURITY 33
+#define JOB_DISPLAY_ORDER_WARDEN 34
+#define JOB_DISPLAY_ORDER_SECURITY_SERGEANT 35 //SKYRAT EDIT ADDITON
+#define JOB_DISPLAY_ORDER_DETECTIVE 36
+#define JOB_DISPLAY_ORDER_SECURITY_OFFICER 37
+#define JOB_DISPLAY_ORDER_JUNIOR_SECURITY_OFFICER 38
+#define JOB_DISPLAY_ORDER_SECURITY_MEDIC 39 //SKYRAT EDIT ADDITON
+#define JOB_DISPLAY_ORDER_PRISONER 40
+#define JOB_DISPLAY_ORDER_BRIGOFF 41 //SKYRAT EDIT ADDITON
```
Comment (FlamingLily) [inline code/__DEFINES/jobs.dm]:
```suggestion
#define JOB_DISPLAY_ORDER_BARTENDER 4
#define JOB_DISPLAY_ORDER_BOTANIST 5
#define JOB_DISPLAY_ORDER_COOK 6
#define JOB_DISPLAY_ORDER_JANITOR 7
#define JOB_DISPLAY_ORDER_CLOWN 8
#define JOB_DISPLAY_ORDER_MIME 9
#define JOB_DISPLAY_ORDER_CURATOR 10
#define JOB_DISPLAY_ORDER_LAWYER 11
#define JOB_DISPLAY_ORDER_CHAPLAIN 12
#define JOB_DISPLAY_ORDER_PSYCHOLOGIST 13
#define JOB_DISPLAY_ORDER_AI 14
#define JOB_DISPLAY_ORDER_CYBORG 15
#define JOB_DISPLAY_ORDER_CHIEF_ENGINEER 16
#define JOB_DISPLAY_ORDER_STATION_ENGINEER 17
#define JOB_DISPLAY_ORDER_ATMOSPHERIC_TECHNICIAN 18
#define JOB_DISPLAY_ORDER_QUARTERMASTER 19
#define JOB_DISPLAY_ORDER_CARGO_TECHNICIAN 20
#define JOB_DISPLAY_ORDER_SHAFT_MINER 21
#define JOB_DISPLAY_ORDER_CHIEF_MEDICAL_OFFICER 22
#define JOB_DISPLAY_ORDER_MEDICAL_DOCTOR 23
#define JOB_DISPLAY_ORDER_PARAMEDIC 24
#define JOB_DISPLAY_ORDER_CHEMIST 25
#define JOB_DISPLAY_ORDER_VIROLOGIST 26
#define JOB_DISPLAY_ORDER_RESEARCH_DIRECTOR 27
#define JOB_DISPLAY_ORDER_SCIENTIST 28
#define JOB_DISPLAY_ORDER_ROBOTICIST 29
#define JOB_DISPLAY_ORDER_GENETICIST 30
#define JOB_DISPLAY_ORDER_EXP_CORPS 31 //SKYRAT EDIT ADDITON
#define JOB_DISPLAY_ORDER_HEAD_OF_SECURITY 32
#define JOB_DISPLAY_ORDER_WARDEN 33
#define JOB_DISPLAY_ORDER_SECURITY_SERGEANT 34 //SKYRAT EDIT ADDITON
#define JOB_DISPLAY_ORDER_DETECTIVE 35
#define JOB_DISPLAY_ORDER_SECURITY_OFFICER 36
#define JOB_DISPLAY_ORDER_JUNIOR_SECURITY_OFFICER 37
#define JOB_DISPLAY_ORDER_SECURITY_MEDIC 38 //SKYRAT EDIT ADDITON
#define JOB_DISPLAY_ORDER_PRISONER 39
#define JOB_DISPLAY_ORDER_BRIGOFF 40 //SKYRAT EDIT ADDITON
#define JOB_DISPLAY_ORDER_BLUESHIELD 41 //SKYRAT EDIT ADDITON - BLUESHIELD
```

This is fuckin' messyyyyyyyy but uh, it's the lesser of two messes I feel

Anyway here's the code done so you can just hit commit changes to revert the blueshield bump up for #5795 

or don't tbh I dunno what course would be least messy, leaving this as is or reverting, merging that, then updating this

**Comment by Gandalf2k15** [comment]
dont care

**Pull request merged**